### PR TITLE
91 Clear input button

### PR DESCRIPTION
**Pull request by mindfulme**
#### What's this PR do?
Adds cross icon along with button functionality and function to clear the current input
#### Where should the reviewer start?
at Input.vue
#### How should this be manually tested?
Basically, you could just test it by typing in the input field in any of the pages
#### Any background context you want to provide?
The clear button appears only when some content is passed to the input. If the input contains an icon then the clear button will not show up as it is mostly unnecessary (i.e. places like copy/paste inputs)
#### What are the relevant tickets?
#91 
#### Screenshots (if appropriate)
screen shot 2018-11-08 at 02 58 36
screen shot 2018-11-08 at 02 58 09

#### URL
Provide a link to your deploy from http://staging-ui.trusty.apasia.tech:8080/
http://staging-ui.trusty.apasia.tech:8080/91-clear-button
#### Questions:
No questions

### CODE REVIEW

**Comment by mindfulme** [comment]
@roma219 Review merge if ok

File: src/assets/svg/cross.svg [hunk 1]
```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
```
Comment (roma219) [inline src/assets/svg/cross.svg]: Where is this icon from?

File: src/components/Input/Input.vue [hunk 2]
```diff
@@ -177,6 +186,9 @@ export default {
     },
     handleIconClick() {
       this.$emit('icon-click')
+    },
+    handleDeleteClick() {
+      this.value = ''
```
Comment (roma219) [inline src/components/Input/Input.vue]: Thats not the right way to change value

**Review by roma219 — CHANGES_REQUESTED** [review]
1) Doesn't work right - try focusing on other fields after
2) image
3) Doesn't look like design
image

File: src/components/Input/Input.vue (hunk 2 above)
Comment (mindfulme) [reply src/components/Input/Input.vue]: @roma219 I have fixed the mutation of the `value` prop

File: src/assets/svg/cross.svg (hunk 1 above)
Comment (mindfulme) [reply src/assets/svg/cross.svg]: This is replaced now by the icon as in zeplin

File: src/components/Input/Input.vue [hunk 3]
```diff
@@ -5,12 +5,12 @@
       'input--has-error': hasError,
       'input--disabled': disabled
     }"
-    class="input">
+    class="input input--delete-icon">
```
Comment (roma219) [inline src/components/Input/Input.vue]: why is this class?

File: src/components/Input/Input.vue [hunk 4]
```diff
@@ -109,7 +118,19 @@ export default {
...
     }
   },
+  data: () => {
+    return {
+      clearInput: false
+    }
+  },
   computed: {
+    getValue() {
+      if (this.clearInput) {
```
Comment (roma219) [inline src/components/Input/Input.vue]: This is awfull.... please don't do it like this. You should not have and 'clearInput' variables

File: src/components/Input/Input.vue [hunk 5]
```diff
@@ -155,6 +176,7 @@ export default {
   },
   methods: {
     handleInput({ target: { value } }) {
+      this.clearInput = false
```
Comment (roma219) [inline src/components/Input/Input.vue]: what is this? It is not needed

File: src/components/Input/Input.vue (hunk 4 above)
Comment (roma219) [reply src/components/Input/Input.vue]: Do you understand that the way you do it you don't actually clear any data, but display empty?

**Review by roma219 — CHANGES_REQUESTED** [review]
Needs a lot of changes. Please remove all you logic for handling clear button - all you need to do is `$emit` empty value and thats it.

File: src/components/Input/Input.vue (hunk 3 above)
Comment (mindfulme) [reply src/components/Input/Input.vue]: got it

File: src/components/Input/Input.vue (hunk 4 above)
Comment (mindfulme) [reply src/components/Input/Input.vue]: got it

File: src/components/Input/Input.vue (hunk 5 above)
Comment (mindfulme) [reply src/components/Input/Input.vue]: got it

**Comment by mindfulme** [comment]
@roma219 review, merge if ok

**Review by roma219 — APPROVED** [review]

**Pull request merged**